slp/trainer/handlers.py

```diff
@@ -68,3 +68,15 @@ def attach(self, trainer: Engine, evaluator: Engine,
             Events.EPOCH_COMPLETED,
             self, evaluator, dataloader,
             validation=validation)
+
+class PeriodicNewbob(object):
+    def __init__(self, period):
+        self.period = period
+
+    def __call__(self, engine: Engine, optimizer):
+        if engine.state.epoch % self.period == 0:
```

Comment:
There is also the classic Newbob that monitors the val loss and if it stops decreasing it does lr/2